mithril-client-cli/src/main.rs

```diff
@@ -139,6 +139,27 @@ impl Args {
 
         Ok(Logger::root(Arc::new(drain), slog::o!()))
     }
+
+    fn parse_with_decorator(
+        decorator: &dyn Fn(Result<Self, clap::error::Error>) -> Result<Self, clap::error::Error>,
+    ) -> Self {
+        let result = decorator(Self::try_parse());
+        match result {
+            Ok(s) => s,
+            Err(e) => e.exit(),
+        }
+    }
+
+    fn handle_deprecated_decorator(
+        args_result: Result<Self, clap::error::Error>,
+    ) -> Result<Self, clap::error::Error> {
+        let styles = Args::command().get_styles().clone();
+        Deprecation::handle_deprecated_commands(
+            args_result,
+            styles,
+            vec![DeprecatedCommand::new("snapshot", "cardano-db")],
```

Comment:
This could be part of the configuration of the `parse_with_decorator` function IMO.

Reply:
Done